common/auto-updater/versions-db-manager.h

```diff
@@ -0,0 +1,138 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <unordered_map>
+#include <functional>
+#include <string>
+#include <vector>
+
+#include "http-downloader.h"
+
+namespace rs2
+{
+
+    class versions_db_manager
+    {
+    public:
+        struct version
+        {
+            int mjor, mnor, patch, build;
+
+            version() : mjor(0), mnor(0), patch(0), build(0) {}
+
+            version(long long ver) : version()
+            {
+                build = ver % 10000;
+                patch = (ver / 10000) % 100;
+                mnor = (ver / 1000000) % 100;
+                mjor = (ver / 100000000) % 100;
+            }
+
+            version(const std::string& str) : version()
+            {
+                std::regex rgx("^(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,5})?$");
+                std::smatch match;
+
+                if (std::regex_search(str.begin(), str.end(), match, rgx) && match.size() >= 4)
+                {
+                    mjor = atoi(std::string(match[1]).c_str());
+                    mnor = atoi(std::string(match[2]).c_str());
+                    patch = atoi(std::string(match[3]).c_str());
+                    if (match.size() == 5)
+                        build = atoi(std::string(match[4]).c_str());
+                }
+            }
+
+            bool operator<=(const version& other) const
+            {
+                if (mjor > other.mjor) return false;
+                if ((mjor == other.mjor) && (mnor > other.mnor)) return false;
+                if ((mjor == other.mjor) && (mnor == other.mnor) && (patch > other.patch)) return false;
+                if ((mjor == other.mjor) && (mnor == other.mnor) && (patch == other.patch) && (build > other.build)) return false;
+                return true;
+            }
+            bool operator==(const version& other) const
+            {
+                return (other.mjor == mjor && other.mnor == mnor && other.patch == patch && other.build == build);
+            }
+
+            bool operator> (const version& other) const { return !(*this < other); }
```

Comment:
I make mistake, but I thought the condition  (!(*this < other); )  is equivalent to >=, not operator >

Reply:
Right, Will be fixed

Reply:
Done